### PR TITLE
[1.40.x-prod] Make shell tests more flexible

### PR DESCRIPTION
**Pull request by radtriste**
and add some prod test scripts

### CODE REVIEW

**Comment by kie-ci1** [comment]
**PR job** `#15` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.40.x-prod/job/pullrequest/job/kogito-images/15/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[Pipeline] error<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-data-index-ephemeral<br/>Build KIE » kogito » 1.40.x-prod » pullrequest » kogito-images.build-image PR #1641 - kogito-swf-devmode: https://github.com/kiegroup/kogito-images/pull/1641 completed: FAILURE<br/>[Pipeline] error<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-swf-devmode<br/>Build KIE » kogito » 1.40.x-prod » pullrequest » kogito-images.build-image PR #1641 - kogito-swf-builder: https://github.com/kiegroup/kogito-images/pull/1641 completed: FAILURE<br/>[Pipeline] error<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-swf-builder<br/>[Pipeline] // parallel<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[INFO] Clean workspace<br/>[Pipeline] cleanWs<br/>[WS-CLEANUP] Deleting project workspace...<br/>[WS-CLEANUP] Deferred wipeout is used...<br/>[WS-CLEANUP] done<br/>[Pipeline] echo<br/>[INFO] Workspace cleaned<br/>[Pipeline] echo<br/>[INFO] Cleanup Maven artifacts<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.m2/repository<br/>[Pipeline] echo<br/>[INFO] .m2/repository cleaned<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.40.x-prod/job/pullrequest/job/kogito-images/15/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by radtriste** [comment]
CI failing due to https://bugzilla.redhat.com/show_bug.cgi?id=2224411

**Comment by radtriste** [comment]
jenkins retest this

**Comment by kie-ci1** [comment]
**PR job** `#16` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.40.x-prod/job/pullrequest/job/kogito-images/16/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-data-index-ephemeral<br/>Build KIE » kogito » 1.40.x-prod » pullrequest » kogito-images.build-image PR #1641 - kogito-swf-builder: https://github.com/kiegroup/kogito-images/pull/1641 completed: FAILURE<br/>[Pipeline] error<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-swf-builder<br/>Build KIE » kogito » 1.40.x-prod » pullrequest » kogito-images.build-image PR #1641 - kogito-swf-devmode: https://github.com/kiegroup/kogito-images/pull/1641 completed: FAILURE<br/>[Pipeline] error<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-swf-devmode<br/>[Pipeline] // parallel<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[INFO] Clean workspace<br/>[Pipeline] cleanWs<br/>[WS-CLEANUP] Deleting project workspace...<br/>[WS-CLEANUP] Deferred wipeout is used...<br/>[WS-CLEANUP] done<br/>[Pipeline] echo<br/>[INFO] Workspace cleaned<br/>[Pipeline] echo<br/>[INFO] Cleanup Maven artifacts<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.m2/repository<br/>[Pipeline] echo<br/>[INFO] .m2/repository cleaned<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.40.x-prod/job/pullrequest/job/kogito-images/16/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.40.x-prod/job/pullrequest/job/kogito-images/16/consoleText<br/>+ tail -n 50<br/>
</details>

**Comment by radtriste** [comment]
artifacts' issues because they do not exist on Nexus ...

File: scripts/logic/test/shell_test.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+script_dir_path="$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)"
+source ${script_dir_path}/env_test.sh $@
+
+echo "---- Pulling image ${image_full_tag} ----"
+docker pull ${image_full_tag}
+
+echo "---- Run behave test for image ${image_id} ----"
```
Comment (MarianMacik) [inline scripts/logic/test/shell_test.sh]: Maybe
```suggestion
echo "---- Run jBang test for image ${image_id} ----"
```
?

**Review by MarianMacik — APPROVED** [review]
Looks good!

**Comment by kie-ci1** [comment]
**PR job** `#18` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.40.x-prod/job/pullrequest/job/kogito-images/18/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[Pipeline] error<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-data-index-ephemeral<br/>Build KIE » kogito » 1.40.x-prod » pullrequest » kogito-images.build-image PR #1641 - kogito-swf-devmode: https://github.com/kiegroup/kogito-images/pull/1641 completed: FAILURE<br/>[Pipeline] error<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-swf-devmode<br/>Build KIE » kogito » 1.40.x-prod » pullrequest » kogito-images.build-image PR #1641 - kogito-swf-builder: https://github.com/kiegroup/kogito-images/pull/1641 completed: FAILURE<br/>[Pipeline] error<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] }<br/>Failed in branch kogito-swf-builder<br/>[Pipeline] // parallel<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[INFO] Clean workspace<br/>[Pipeline] cleanWs<br/>[WS-CLEANUP] Deleting project workspace...<br/>[WS-CLEANUP] Deferred wipeout is used...<br/>[WS-CLEANUP] done<br/>[Pipeline] echo<br/>[INFO] Workspace cleaned<br/>[Pipeline] echo<br/>[INFO] Cleanup Maven artifacts<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.m2/repository<br/>[Pipeline] echo<br/>[INFO] .m2/repository cleaned<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/1.40.x-prod/job/pullrequest/job/kogito-images/18/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Pull request merged**